src/ReinforcementLearningCore/src/core/hooks.jl

```diff
@@ -138,27 +140,33 @@ function (hook::TotalRewardPerEpisode)(
     hook.reward = 0
 end
 
-function (hook::TotalRewardPerEpisode{Bool})(
-    ::PostExperimentStage,
-    agent,
-    env,
-)
-    println(
-        lineplot(
+function Base.show(io::IO, hook::TotalRewardPerEpisode{true, F}) where {F<:Number}
+    if length(hook.rewards) > 0
+        println(io, lineplot(
             hook.rewards,
             title="Total reward per episode",
             xlabel="Episode",
             ylabel="Score",
-        ),
-    )
+        ))
+    else
+        println(io, typeof(hook))
+    end
+end
+
+function (hook::TotalRewardPerEpisode{true, F})(
+    ::PostExperimentStage,
+    agent,
+    env,
+) where {F<:Number}
+    display(hook)
```

Comment:
I like the use of display here because some day we could let the plotting backend be user-configurable to instead use Plots or Makie.